marklogic-spark-connector/src/main/java/com/marklogic/spark/writer/NewDocumentProcessor.java

```diff
@@ -22,25 +26,35 @@ public class NewDocumentProcessor {
 
     private final Tika tika;
     private final TextSplitter textSplitter;
+    private final TextClassifier textClassifier;
 
-    public NewDocumentProcessor(Tika tika, TextSplitter textSplitter) {
+    public NewDocumentProcessor(Tika tika, TextSplitter textSplitter, TextClassifier textClassifier) {
         this.tika = tika;
         this.textSplitter = textSplitter;
+        this.textClassifier = textClassifier;
     }
 
-    public DocBuilder.DocumentInputs processDocument(DocBuilder.DocumentInputs inputs) {
+    public void processDocument(DocBuilder.DocumentInputs inputs) {
+        final String uri = inputs.getInitialUri();
         if (tika != null && inputs.getContent() instanceof BytesHandle) {
             extractText(inputs);
         }
         if (textSplitter != null) {
+            applySplitter(inputs);
+        }
+
+        // Not sure if we should run the classifier if the user specifies a splitter. We may need options to control
+        // whether this happens or not.
+        if (textClassifier != null) {
             if (inputs.getExtractedText() != null) {
-                StringHandle content = new StringHandle(inputs.getExtractedText());
-                inputs.setChunks(textSplitter.split(inputs.getInitialUri(), content));
+                byte[] classification = textClassifier.classifyTextToBytes(uri, inputs.getExtractedText());
+                inputs.setClassificationResponse(classification);
             } else {
-                inputs.setChunks(textSplitter.split(inputs.getInitialUri(), inputs.getContent()));
+                String content = HandleAccessor.contentAsString(inputs.getContent());
+                byte[] classification = textClassifier.classifyTextToBytes(uri, content);
```

Comment:
Just FYI - if we know the content is XML, we can return a Document from the classificationClient, which could be a performance improvement.

Reply:
Looking at the S4 source code - it's just doing the same thing we are, which is feeding the byte[] into a DOM DocumentBuilder. 

I was thinking more along the lines of checking for a byte[] and avoiding the cast to a string if we have a BytesHandle. But just getting things working for now.